src/main/java/org/shanerx/tradeshop/listeners/ShopTradeListener.java

```diff
@@ -174,7 +174,7 @@ public void onBlockInteract(PlayerInteractEvent e) {
 
     private boolean tradeAllItems(Shop shop, int multiplier, Action action, Player buyer) {
         ArrayList<ItemStack> costItems = new ArrayList<>(), productItems = new ArrayList<>();
-        Inventory shopInventory = shop.getChestAsSC().getInventory();
+        Inventory shopInventory = shop.hasStorage() ? shop.getChestAsSC().getInventory() : null;
```

Comment:
This is used when checking the inventory of the shop in the #getItems method, it can be null for iTrade shops hence the lamda if

Used on lines 214 & 236